src/Dialog/formComponents/validator/versionConstraintFormat.test.js

```diff
@@ -0,0 +1,32 @@
+/* global it, describe */
+import chai from 'chai';
+
+import versionConstraintFormat from './versionConstraintFormat';
+
+chai.should();
+
+describe('versionConstraintFormat', () => {
+  it('should return true for empty string', () => {
+    const testValue = '';
+
+    versionConstraintFormat(testValue).should.equal(true);
+  });
+
+  it('should return true for particular constraints', () => {
```

Comment:
This comment actually contains an anwers for questions about `null` and empty string.
When a schema property contains only type `string`, form/dialog allows for an empty string because this field is not marked as a required.
Gohan WebUI has a separated "module" for handling `null` and empty strings. When `type` contains `null`, an empty string is converted to appropriate value or undefined (at this moment I don't remember the exact action).
I suggest that property should have `"type": ["string", "null"]` in it's schema.

Reply:
To sum up: validator checks only non-empty strings. Empty strings/required/null are checked by the module for validating consistency with schema.